source/MaterialXRenderGlsl/TextureBaker.cpp

```diff
@@ -124,6 +132,13 @@ void TextureBaker::bakeShaderInputs(ConstShaderRefPtr shaderRef, GenContext& con
         if (output)
         {
             FilePath filename = FilePath(outputFolder / generateTextureFilename(output, shaderRef->getName(), udim));
+
+            if (connectsToNormalMapNode(output))
+            {
+                ElementPtr normalMapNode = output->getParent()->getChild(output->getAttribute("nodename"));
+                output->setAttribute("nodename", normalMapNode->getChild("in")->getAttribute("nodename"));
```

Comment:
Here we can use built-in methods Output::setNodeName, Node::getInput, and Input::getNodeName, in order to avoid these hardcoded strings.